src/delimited_index_connection.cc

```diff
@@ -209,8 +214,13 @@ delimited_index_connection::delimited_index_connection(
         break;
       }
     }
-    write_fut = std::async(
-        [&, i, sz] { std::fwrite(buf[i].data(), sizeof(char), sz, out); });
+    total_bytes_to_write += sz;
```

Comment:
It feels nice to give the user a sense of how much space appears to be needed.

Reply:
It is not, now that I look at `total_read`. Removing `total_bytes_to_write` here and on the fixed width side, in favor of using `total_read`.